ee/control/client_http.go

```diff
@@ -40,77 +61,140 @@ func NewControlHTTPClient(logger log.Logger, addr string, client *http.Client, o
 	return c, nil
 }
 
-func (c *HTTPClient) Get(hash string) (data io.Reader, err error) {
-	verb, path := "GET", fmt.Sprintf("/api/v1/control/%s", hash)
+func (c *HTTPClient) GetConfig() (io.Reader, error) {
+	challengeReq, err := http.NewRequest(http.MethodGet, c.url("/api/agent/config").String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not create challenge request: %w", err)
+	}
+
+	challenge, err := c.do(challengeReq)
+	if err != nil {
+		return nil, fmt.Errorf("could not make challenge request: %w", err)
+	}
+
+	configReq, err := http.NewRequest(http.MethodPost, c.url("/api/agent/config").String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not create config request: %w", err)
+	}
+	configReq.Header.Set(HeaderChallenge, string(challenge))
+	configReq.Header.Set("Content-Type", "application/json")
+	configReq.Header.Set("Accept", "application/json")
+
+	// Calculate first signature
+	localDbKeys := agent.LocalDbKeys()
+	if localDbKeys.Type() != "noop" {
+		key1, err := keyHeaderValue(localDbKeys)
+		if err != nil {
+			return nil, fmt.Errorf("could not get key header from local db keys: %w", err)
+		}
+		sig1, err := signatureHeaderValue(localDbKeys, challenge)
+		if err != nil {
+			return nil, fmt.Errorf("could not get signature header from local db keys: %w", err)
+		}
+		configReq.Header.Set(HeaderKey, key1)
+		configReq.Header.Set(HeaderSignature, sig1)
+	}
+
+	// Calculate second signature if available
+	hardwareKeys := agent.HardwareKeys()
+	if hardwareKeys.Type() != "noop" {
```

Comment:
Might need a timeout on this. 

@James-Pickett I wonder if we should bake some kind of helper into krypto. Something like `echelper.Sign(signer1, signer2) (sig1, sig2, error)` or `echelper.Sign(signers ...crypto.Signers) ([]sigs, error)`

Reply:
I've exposed a func in krypto echelper that does signing with time out, accepting duration and interval as parameters. I played around with what @directionless suggested, but it felt weird because ...

what if it returns an error?
which one failed? 
I accept it if only one fails, but if I get back an array of signatures, how do I know which sig matches which key?

so I thought about having custom error messages that says which index of the given sigs failed ... but I feel like that added complexity is > the complexity of just calling sign twice.

I'm open to discussion =)

Reply:
I _suspect_ the common case is going to be consistent:
1. first sig is local, hard fail for that
2. second sig is hardware. Probably don't fail. maybe expose option.

I don't know what's more idiomatic. Make the callers do the work, or write a helper. I'm happy to roll with whatever.

Reply:
I think an uncommon success condition like "1st of n" should be the responsibility of the consumer .... If I'm using a library and I see a func `SignWithTimeout` that accepts multiple signers I expect an error if any of them fails. I know we are the only consumers of this library, but it just feels weird to me =). So if you good with whatever, lets let launcher handle it.